### PR TITLE
Unit tests for BatchActions component (part2)

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
followup PR of PR #5282
adds more tests in BatchActions component

## Related Issue
- part of #5234

## Motivation and Context
unit test coverage

## How Has This Been Tested?
- local

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js [hunk 1]
```diff
@@ -22,105 +49,212 @@ describe('Batch Actions component', () => {
...
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display accept button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(acceptButton.exists()).toBeTruthy()
+              expect(acceptButton.text()).toContain('Accept')
+              expect(declineButton.exists()).toBeFalsy()
+            })
+            it('should call "acceptShares" when accpet button is clicked', async () => {
+              await acceptButton.vm.$emit('click')
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:141]: can you use trigger here and other usages?
Directly emitted *click* on the button does not verify whether the call is initiated after the button click.

Comment (saw-jan) [reply packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:141]: I have stubbed `ODS` components and used `shallowMount` as per the comments I received previously ([see](https://github.com/owncloud/web/pull/5282#discussion_r653584960))
click cannot be triggered as `.trigger('click')` on stubbed components so have used `emit`

Would you want me to use `mount`?

Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:141]: Imo mount should be used while necessary. For test which involves see emitting should use mount to mount components

File: packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js [hunk 2]
```diff
@@ -1,134 +1,264 @@
...
+              acceptButton = wrapper.find(elSelector.acceptButton)
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display accept button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(acceptButton.exists()).toBeTruthy()
+              expect(acceptButton.text()).toContain('Accept')
+              expect(declineButton.exists()).toBeFalsy()
+            })
+            it('should call "acceptShares" when accpet button is clicked', async () => {
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:136]: it should not display decline button

File: packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js [hunk 3]
```diff
@@ -1,134 +1,264 @@
...
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display decline button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(declineButton.exists()).toBeTruthy()
+              expect(declineButton.text()).toContain('Decline')
+              expect(acceptButton.exists()).toBeFalsy()
+            })
+            it('should call "declineShares" when decline button is clicked', async () => {
+              await declineButton.trigger('click')
```
Comment (kiranparajuli589) [inline packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:160]: it should not display accept button

**Review by kiranparajuli589 — COMMENTED** [review]
Would it be possible to add these tests here as well?

File: packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js (hunk 2 above)
Comment (saw-jan) [reply packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:136]: the function implementation is mocked so it's not possible to check for the changes

File: packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js (hunk 3 above)
Comment (saw-jan) [reply packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js:160]: the function implementation is mocked so it's not possible to check for the changes

**Comment by kiranparajuli589** [comment]
lgtm 👍🏼

**Review by individual-it — CHANGES_REQUESTED** [review]
please squash the commits

**Review by individual-it — APPROVED** [review]

**Pull request merged**